components/views/dropdown-menu-view.js

```diff
@@ -0,0 +1,262 @@
+define([
+        'backbone',
+        'jquery',
+        'underscore',
+        'text!../templates/dropdown.underscore'
+    ],
+    function( Backbone, $, _, DropdownTpl ) {
+        'use strict';
+
+        /**
+         *  Dropdown Menu View which is fully accessible
+         *
+         *  Initialize the view by passing in the following attributes
+         *  className: 'space separated string of classes for element',
+         *  model: with the following attributes (example values added)
+         *      main: {
+         *          text: 'username',
+         *          url: 'dashboard'
+         *      },
+         *      button_label: 'User options dropdown',
+         *      items: [
+         *          {
+         *              text: 'Dashboard',
+         *              url: 'dashboard'
+         *          }, {
+         *              text: 'Account',
+         *              url: 'account_settings'
+         *          }, {
+         *              text: 'Profile',
+         *              url: 'learner_profile'
+         *          }, {
+         *              text: 'Sign Out',
+         *              url: 'logout'
+         *          }
+         *      ]
+         *  parent: 'selecter for parent element that will be replaced with dropdown menu'
+         *
+         */
+        return Backbone.View.extend({
+            tpl: _.template( DropdownTpl ),
+
+            events: {
+                'click .js-dropdown-button': 'clickOpenDropdown',
+                'click a': 'analyticsLinkClick',
+                'keydown': 'viewKeypress'
+            },
+
+            menu: '.dropdown-menu',
+
+            key: {
```

Comment:
Can we put these into a common utility class? I'm sure we're going to use these a lot. In edx-platform we use the JQuery UI constants, but obviously we don't want to bring that monstrosity in...